rewrite-core/src/main/java/org/openrewrite/text/FindAndReplace.java

```diff
@@ -134,7 +134,8 @@ public Tree visit(@Nullable Tree tree, ExecutionContext ctx) {
                 if (!matcher.find()) {
                     return sourceFile;
                 }
-                String newText = matcher.replaceAll(replace);
+                String replacement = Boolean.TRUE.equals(regex) ? replace : replace.replaceAll("\\$", "\\\\\\$");
```

Comment:
```suggestion
                String replacement = replace;
                if (!Boolean.TRUE.equals(regex)) {
                    replacement = replacement.replace("$", "\\$");
                }```
WDYT to have it more similar to the similar issue on line 117? Also, use replace instead of replaceAll since we do not need a regex?

Reply:
Good suggestion, thanks! Let's merge as soon as the tests pass.